backend/src/LibExecution/Interpreter.fs

```diff
@@ -652,21 +652,24 @@ let rec eval'
 
 
     | EIf(id, cond, thenbody, elsebody) ->
+      let elsebodyUnwrapped = Option.defaultValue (EUnit id) elsebody
       match! eval state tat st cond with
       | DBool false ->
         do! preview tat st thenbody
-        return! eval state tat st elsebody
+        match elsebody with
+        | None -> return DUnit
+        | Some eb -> return! eval state tat st eb
       | DBool true ->
         let! result = eval state tat st thenbody
-        do! preview tat st elsebody
+        do! preview tat st elsebodyUnwrapped
```

Comment:
Using the id for elseBody will save `()` as the return value for the `if` statement. I don't think that's right for this branch.

Reply:
I'm not sure that I understand what is the right option here. Could you please to add details?

Reply:
We make calls to `preview` so that we have analysis results for the path that wasn't executed.

That is, if we have

```
if true
  let x = 5
  6
else
  let y = "str"
  7
```

then we'd like to know that y would have been 7 if we had executed the other branch. That's what `preview` handles (it stores the result of expressions using `NotExecuted`).

If there is no else body, there is no preview to execute. Previewing using elseBodyUnwrapped adds an extra analysis result using `id` (which is the `id` for the `if`, so that's wrong). This will overwrite the result of the `if` statement with `NotExecuted DUnit`.

So when `cond` is true and there's no `elseBody`, there's nothing to preview.

Reply:
@pbiggar thanks for explanation! As I've understood, we use `preview` to show value (7) on canvas for `NotExecuted` branch (of course, if the branch exists). Hope I understand correctly how to apply preview for `elseBody`, please check the last commit.